src/device.rs

```diff
@@ -74,46 +85,25 @@ impl Device {
     }
 
     pub async fn flush(&mut self) -> Result<(), Box<dyn Error>> {
-        const MTU_SIZE: usize = 20;
-
-        while !self.tx_buffer.is_empty() {
-            let mut buf = Vec::with_capacity(MTU_SIZE);
-            for _ in 0..MTU_SIZE {
-                if let Some(byte) = self.tx_buffer.pop_front() {
-                    buf.push(byte);
-                } else {
-                    break;
-                }
+        let chunks = self.tx_buffer.chunks(20);
```

Comment:
Nice solution! I always forget about the Vec-convenience methods. Just one small suggestion to avoid magic numbers.
```suggestion
        const MTU_SIZE: usize = 20;
        let chunks = self.tx_buffer.chunks(MTU_SIZE);
```